concourse/steps/release.py

```diff
@@ -334,8 +343,51 @@ def _push_tag(tag):
         else:
             raise NotImplementedError
 
+        if self.publishing_policy is ReleaseCommitPublishingPolicy.TAG_AND_MERGE_BACK:
+            def create_merge_commit(head: git.types.Commit_ish):
+                merge_base = self.git_helper.repo.merge_base(
+                    head,
+                    self.context().release_commit,
+                )
+
+                self.git_helper.repo.index.merge_tree(
+                    rhs=self.context().release_commit,
+                    base=merge_base,
+                )
+
+                return self.git_helper.index_to_commit(
+                    message=f'Merge release-commit from tag {self.release_tag}',
+                    parent_commits=(
+                        head,
+                        self.context().release_commit,
+                    ),
+                )
+
+            try:
+                head = self.git_helper.fetch_head(
+                    f'refs/heads/{self.repository_branch}'
+                ) # this is safe as release-tag is created already
+
+                merge_commit = create_merge_commit(head)
+                self.context().merge_release_back_to_default_branch_commit = merge_commit # pass to other steps
```

Comment:
nit: `context()` is always used to pass data between other steps - no need to add this as a comment